scripts/colonycorefu.lua

```diff
@@ -3,76 +3,56 @@ require "/scripts/kheAA/transferUtil.lua"
 local bonusHappiness
 -- This is a modified well script. wellsDrawing is being used to represent how many tenants are active under this colony core.
 local rentTimer
-local rentGoal
+local rentTime
+local offlineTicks=0
+
+-- This object keep work even if it's unloaded. The trick is simple
+-- It simply remembers the time it was unloaded. When is loaded it will try to compensate all missed ticks.
+-- It will compensate maximum 24 hours of work.
 
 function init()
 	transferUtil.loadSelfContainer()
 	wellRange=config.getParameter("wellRange",128)
-	rentGoal=config.getParameter("productionTime",150)
+	rentTime=config.getParameter("productionTime",150)
+	wellSlots=config.getParameter('wellslots')
+	maxOfflineSeconds=config.getParameter("maxOfflineSeconds",86400)--86400 1 day
 	bonusHappiness = 10
 	rentTimer = 0
 	wellInit()
 	setDesc()
+
+	--figuring out how many ticks we missed
+	local leftTime = config.getParameter("leftTime")
+	if leftTime then
+		local diff=os.time() - leftTime
+		if diff>0 then
+			diff=math.min(math.floor(diff/rentTime),math.floor(maxOfflineSeconds/rentTime))
+			offlineTicks=diff-1
```

Comment:
Need a sanity check that `offlineTicks` is not negative, e.g. `math.max(0, diff - 1)`

Reply:
diff can not be negative since we already check if it's greater than zero. And diff can not be float since both os.time() and leftTime are in POSIX format (integer amount of second)

Reply:
You are setting `diff` to a different value after the `diff>0` check, and it can become 0 there.

Reply:
OK i'll add additional check tomorrow, just in case

Reply:
`offlineTicks=math.max(0, diff - 1)`